.github/workflows/smart_contracts.yaml

```diff
@@ -157,7 +164,10 @@ jobs:
     steps:
       - uses: "actions/checkout@v4"
       - if: |-
-          github.ref_type == 'tag'
+          github.ref_type == 'tag' || (
+            github.event_name == 'workflow_dispatch' &&
+              inputs.check-dependencies-version
+          )
```

Comment:
may we reuse `check-dependencies-versions`?

Reply:
I should be able to. There were some exceptions to where the `env` context can be used, but it should work here.